scene/resources/mesh_data_tool.h

```diff
@@ -49,6 +56,7 @@ class MeshDataTool : public RefCounted {
 		Vector<int> edges;
 		Vector<int> faces;
 		Variant meta;
+		CustomVertexData custom[Mesh::ARRAY_CUSTOM_COUNT];
```

Comment:
I would use Color here also, when providing arrays Godot will for the most part give or take this. You can keep the format flat somewhere so godot properly reencodes it when setting back the array.